### PR TITLE
ligthen process with makefile

### PR DESCRIPTION
**Pull request by MalloZup**
- make all basic operation simplier
- improve doc with this

### CODE REVIEW

**Comment by MalloZup** [comment]
this is ready for review

**Comment by monstermunchkin** [comment]
Could we rename GNUmakefile to Makefile? That is more common, and even GNU recommends using Makefile if it's not specific to GNU make.

**Comment by MalloZup** [comment]
@monstermunchkin thx  done. I took inspiration from this: https://github.com/terraform-providers/terraform-provider-google/blob/master/GNUmakefile.

Since it has gone a while last time i used a Makeifle, i was thinking that they changed some conventions :rofl:

File: doc/CONTRIBUTING.md [hunk 1]
```diff
@@ -5,23 +5,30 @@ Look at open issues, especially with label:
...
+
+Go in your build dir ```cd $GOPATH/src/github.com/dmacvicar/terraform-provider-libvirt```,
+then  use
+
+```console
+make testacc
+```
+This will build the plugin , performing basic checks (gofmt) and run the acceptance test on your local machine.
+
+
```
Comment (monstermunchkin) [inline doc/CONTRIBUTING.md]: This section is unnecessary.

File: doc/CONTRIBUTING.md [hunk 2]
```diff
@@ -5,23 +5,30 @@ Look at open issues, especially with label:
...
 # Contribute
 
-1. [Build the plugin from source](https://github.com/dmacvicar/terraform-provider-libvirt#building-from-source)
-2. If you have issues, check out the [troubleshooting](https://github.com/dmacvicar/terraform-provider-libvirt/blob/master/doc/TROUBLESHOOTING.md)
-3. Do your code (reference issue on your pr if you fix them) [Closing issues keywords](https://help.github.com/articles/closing-issues-using-keywords/)
-4. Run the acceptance tests.
+1. If you have issues, check out the [troubleshooting](https://github.com/dmacvicar/terraform-provider-libvirt/blob/master/doc/TROUBLESHOOTING.md)
+2. Do your code (reference issue on your pr if you fix them) [Closing issues keywords](https://help.github.com/articles/closing-issues-using-keywords/)
+3. Use the Makefile workflow before submit the PR (building, tests).
```
Comment (monstermunchkin) [inline doc/CONTRIBUTING.md]: 3. Test your code by running the acceptance tests
4. Create a PR

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,22 @@
+default: build
+
+build: gofmtcheck
+	go install
```
[the remark this answers is not in the document]
Comment (MalloZup) [reply Makefile]: i was doubting about this, i just took inspiration from other upstream project.
Ok we can create an extra target no pb

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+testacc: build
```
[the remark this answers is not in the document]
Comment (MalloZup) [reply Makefile]: imho, i have think this makefile for developers.
So you want to tests your builded code, that's why i added the dependency.

File: doc/CONTRIBUTING.md (hunk 2 above)
Comment (MalloZup) [reply doc/CONTRIBUTING.md]: done

**Comment by MalloZup** [comment]
@monstermunchkin thx i added it.

**Comment by MalloZup** [comment]
@monstermunchkin  i think i adressed all your comments. thank you for your review. :boat:

File: doc/CONTRIBUTING.md (hunk 2 above)
Comment (monstermunchkin) [reply doc/CONTRIBUTING.md]: Not done yet.

**Comment by monstermunchkin** [comment]
Not all comments have been addressed yet.

**Comment by MalloZup** [comment]
@monstermunchkin  ok yop i missed to remove 1 line. now is fixed

File: doc/CONTRIBUTING.md [hunk 5]
```diff
@@ -5,23 +5,29 @@ Look at open issues, especially with label:
...
 
+### Makefile devel workflow.
+
+Go in your build dir ```cd $GOPATH/src/github.com/dmacvicar/terraform-provider-libvirt```,
+then  use
+
+```console
+make test
+```
+
```
Comment (monstermunchkin) [inline doc/CONTRIBUTING.md]: Please remove this section as well. It's unnecessary.

Comment (MalloZup) [reply doc/CONTRIBUTING.md]: yop, done

**Comment by monstermunchkin** [comment]
How about this:
```text
# Where to start?

Look at open issues, especially with label:

- [junior job](https://github.com/dmacvicar/terraform-provider-libvirt/issues?q=is%3Aissue+is%3Aopen+label%3A%22junior+job%22), 
- [help wanted](https://github.com/dmacvicar/terraform-provider-libvirt/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22)

### Workflow

1. If you have issues, check out the [troubleshooting](https://github.com/dmacvicar/terraform-provider-libvirt/blob/master/doc/TROUBLESHOOTING.md)
2. Do your code ([reference issue(s)](https://help.github.com/articles/closing-issues-using-keywords/) on your PR if you fix them)
3. Test your code by running the acceptance tests (`make test`)
4. Create a PR

### Writing acceptance tests

Take a look at Terraform's docs about [writing acceptance tests](https://github.com/hashicorp/terraform/blob/master/.github/CONTRIBUTING.md#writing-an-acceptance-test).
```

**Comment by MalloZup** [comment]
@monstermunchkin  copy and pasted the suggestion :dog:

**Pull request merged**